WordPress/gutenberg
# Add `lang` and `dir`  attributes to text-formatting tools

## What?

This PR adds the ability to select the `lang` and `dir` attributes, So that users can make parts of their text in a different language. This helps assistive technologies like screen-readers to properly understand what language the text is in, and use the proper pronunciation when reading the text.

## Why?
Because not all text is in English and there are lots of sites that use more than one languages. This addition will help content writers to produce more accessible content.

## How?
We started exploring ways to do that with @afercia, @carolinan and @SergeyBiryukov, but we later found that @audrasjb already built a plugin for that in https://github.com/audrasjb/language-attribute-for-gutenberg
This PR is a port of that plugin to Gutenberg, huge props to @audrasjb for all the work there! ❤️ 
We fixed some deprecation notices, and adapted the code a bit for use in Gutenberg, but other than that it's an almost verbatim copy of that code.

## Testing Instructions
1. Add some text and then select it.
3. Go to the text tools, and click on the language icon
4. Add a language like `fr`, `it` etc, and select if the language is LTR or RTL
5. Verify that the markup (both in the editor and the frontend has the right `lang` and `dir` attributes
6. Verify that you can remove the attributes by clicking on the language selector again

## Screenshots or screencast

The screencast below shows selecting and deselecting the language, and how that applies to the markup.

https://user-images.githubusercontent.com/588688/233589432-2b8b3c65-e9e4-42cf-a1d4-670ead3c7dab.mov

--------

Props @audrasjb, @afercia, @carolinan, @SergeyBiryukov, @aristath
